client/src/context/useKanbanContext.tsx

```diff
@@ -50,57 +89,64 @@ export const KanbanProvider: FunctionComponent = ({ children }): JSX.Element =>
     draggableId: string,
   ): ICard[] => {
     const cardsCopy = [...cards];
-    const cardIndex = cardsCopy.findIndex((card: ICard) => card.id === draggableId);
+    const cardIndex = cardsCopy.findIndex((card: ICard) => card._id === draggableId);
     if (cardIndex > -1) {
       const [card] = cardsCopy.splice(source.index, 1);
       cardsCopy.splice(destination.index, 0, card);
+
       return cardsCopy;
     }
+
     return cards;
   };
 
   const addCard = (card: ICard): boolean => {
     if (card.name === '') {
       updateSnackBarMessage('Please enter a card name');
+
       return false;
     }
 
     const columnsCopy = cloneDeep(columns);
-    const columnIndex = columnsCopy.findIndex((col) => col.id === card.columnId);
+    const columnIndex = columnsCopy.findIndex((col) => col._id === card.columnId);
     if (columnIndex > -1) {
       const columnCopy = cloneDeep(columns[columnIndex]);
       columnCopy.cards.push(card);
       columnsCopy[columnIndex] = columnCopy;
+      activeBoard.columns = columnsCopy;
+
       setColumns(columnsCopy);
+
+      updateBoard(activeBoard);
+
       return true;
     }
+
     return false;
   };
 
-  const setOpenCard = (card: ICard): void => {
-    setFocusedCard(card);
-  };
+  const setOpenCard = (card: ICard): void => setFocusedCard(card);
+
   const resetOpenCard = (): void => setFocusedCard(null);
 
   const getColumnById = (columnId: string): IColumn | null => {
-    if (!columnId) return null;
```

Comment:
Removed null return case as columnId should always be present according to its implementation